types/index.d.ts

```diff
@@ -0,0 +1,227 @@
+     * when touch is made outside the options.catchDistance the previous joystick is destroyed and a new one is created.
+     * cannot be multitouch.
+     *
+     * ### 'static':
+     * a joystick is positioned immediately at options.position.
+     * one joystick per zone.
+     * each new touch triggers a new direction.
+     * cannot be multitouch.
+     */
+    mode?: 'dynamic' | 'semi' | 'static';
+
+    /**
+     * Defaults to `true`
+     *
+     * Reset the joystick’s position when it enters the rest state.
+     */
+    restJoystick?: boolean;
+
+    /**
+     * Defaults to `0.5`
+     * The opacity to apply when the joystick is in a rest position.
+     */
+    restOpacity?: number;
+
+    /**
+     * Defaults to `200`
+     *
+     * This is only useful in the semi mode, and determine at which distance we recycle the previous joystick.
+     */
+    catchDistance?: number;
+
+    /**
+     * Defaults to `false`
+     *
+     * Locks joystick’s movement to the x (horizontal) axis
+     */
+    lockX?: boolean;
+
+    /**
+     * Defaults to `false`
+     *
+     * Locks joystick’s movement to the y (vertical) axis
+     */
+    lockY?: boolean;
+}
+
+export interface Position {
+    x: number;
+    y: number;
+}
+
+export interface JoystickOutputData {
+    angle: {
+        degree: number;
+        radian: number;
+    };
+    direction: {
+        angle: string;
+        x: string;
+        y: string;
+    };
+    distance: number;
+    force: number;
+    identifier: number;
+    instance: Joystick;
+    position: Position;
+    pressure: number;
+}
+
+export interface EventData {
+    type: string;
+    target: any;
+}
+
+export class JoystickManager {
+    static create(options?: JoystickManagerOptions): JoystickManager;
+
+    on(
+        type: string | string[],
+        handler: (evt: EventData, data: JoystickOutputData) => void
+    ): void;
+    off(
+        type: string | string[],
+        handler: (evt: EventData, data: JoystickOutputData) => void
+    ): void;
+    get(identifier: number): Joystick;
+    destroy(): void;
+}
+
+export interface Joystick {
+    on(
+        type: string | string[],
+        handler: (evt: EventData, data: JoystickOutputData) => void
+    ): void;
+    off(
+        type: string | string[],
+        handler: (evt: EventData, data: JoystickOutputData) => void
+    ): void;
+    el: HTMLElement;
+    show(cb?: () => void): void;
+    hide(cb?: () => void): void;
+    add(): void;
+    remove(): void;
+    destroy(): void;
+    identifier: number;
+    trigger(type: string | string[], handler: (evt: EventData, data: any) => void): void;
+    position: Position;
+    frontPosition: Position;
+    ui: {
+        el: HTMLElement;
+        front: HTMLElement;
+        back: HTMLElement;
+    };
+    options: JoystickManagerOptions;
+}
+
+/**
+ * A JavaScript library for creating vanillaJS virtual joysticks, for touch capable interfaces.
+ */
+declare module 'nipplejs' {
```

Comment:
Looking at the source code, this accurately reflects the JS module, and also is what works for me consuming the TypeScript definition.

Reply:
This for sure works with `import * as nipplejs from 'nipplejs'` for me. The other example did not.

Good questions. I think if the JavaScript were exporting a different module format or different "module" object were being returned from the IIFE it might allow other import options. Honestly I mostly consume .d.ts, not create them, so I'm no pro either. 

I made the assumption that I should only edit .d.ts and not make JS changes for this PR, but I guess if I broke anything, the test suite should tell me. I don't really have more time to spend experimenting, atm.